oss-internship-2020/libtiff/test/raw_decode.cc

```diff
@@ -0,0 +1,244 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <array>
+#include <cstring>
+
+#include "helper.h"  // NOLINT(build/include)
+#include "tiffio.h"  // NOLINT(build/include)
+
+namespace {
+
+using ::sapi::IsOk;
+using ::testing::Eq;
+using ::testing::IsFalse;
+using ::testing::IsTrue;
+using ::testing::NotNull;
+
+struct ChannelLimits {
```

Comment:
This seems to be defined in 2 places.
Maybe put in a common header.
Also perhaps define a helper `bool Check(uint8_t red, uint8_t green, uint8_t blue, uint8_t alpha)` inside the class.

Reply:
one in the example and one in the test. I'm not sure if I should make a common header for them, should I?